src/hooks/test/use-tab-key-navigation-test.js

```diff
@@ -0,0 +1,245 @@
+      bubbles: true,
+      cancelable: true,
+      ...eventOptions,
+    });
+    act(() => {
+      findElementByTestId('toolbar').dispatchEvent(event);
+    });
+    return event;
+  }
+
+  function currentItem() {
+    return document.activeElement.innerText;
+  }
+
+  it('should move focus and tab stop between elements when tab keys are pressed', () => {
+    const forwardKey = 'Tab';
+    const backKey = { key: 'Tab', shiftKey: true };
+
+    renderToolbar();
+
+    const steps = [
+      // Test navigating forwards.
+      [forwardKey, 'Italic'],
+      [forwardKey, 'Underline'],
+      [forwardKey, 'Help'],
+
+      // Test that navigation wraps to start.
+      [forwardKey, 'Bold'],
+
+      // Test that navigation wraps to end.
+      [backKey, 'Help'],
+
+      // Test navigating backwards.
+      [backKey, 'Underline'],
+      [backKey, 'Italic'],
+      [backKey, 'Bold'],
+    ];
+
+    for (let [key, expectedItem] of steps) {
+      pressKey(key);
+
+      const currentElement = document.activeElement;
+      assert.equal(currentElement.innerText, expectedItem);
+
+      const toolbarButtons = container.querySelectorAll('a,button');
+      for (let element of toolbarButtons) {
+        if (element === currentElement) {
+          assert.equal(element.tabIndex, 0);
+        } else {
+          assert.equal(element.tabIndex, -1);
+        }
+      }
+    }
+  });
+
+  [
+    {
+      key: 'Tab',
+      shouldHandle: true,
+    },
+    {
+      key: { key: 'Tab', shiftKey: true },
+      shouldHandle: true,
+    },
+    {
+      key: 'Space',
+      shouldHandle: false,
+    },
+  ].forEach(({ key, shouldHandle }) => {
+    it('should stop keyboard event propagation if event is handled', () => {
+      renderToolbar();
+
+      const handleKeyDown = sinon.stub();
+      container.addEventListener('keydown', handleKeyDown);
+
+      const event = pressKey(key);
+      assert.equal(
+        event.defaultPrevented,
+        shouldHandle,
+        `${key} defaultPrevented`
+      );
+      assert.equal(handleKeyDown.called, !shouldHandle, `${key} propagated`);
+      handleKeyDown.resetHistory();
+    });
+  });
+
+  it('should skip hidden elements', () => {
+    renderToolbar();
+    findElementByTestId('bold').focus();
+    findElementByTestId('italic').style.display = 'none';
+
+    pressKey('Tab');
+
+    assert.equal(currentItem(), 'Underline');
+  });
+
+  it('should skip disabled elements', () => {
+    renderToolbar();
+    findElementByTestId('bold').focus();
+    findElementByTestId('italic').disabled = true;
+
+    pressKey('Tab');
+
+    assert.equal(currentItem(), 'Underline');
+  });
+
+  it('should not respond to keys if hook is not enabled', () => {
+    renderToolbar({
+      enabled: false,
+    });
+    ['Tab', { key: 'Tab', shiftKey: true }].forEach(key => {
+      findElementByTestId('bold').focus();
+      pressKey(key);
+      assert.equal(currentItem(), 'Bold');
+    });
+  });
```

Comment:
This hook can be "enabled" or "disabled", which is a feature that `useArrowKeyNavigation` does not have.